api/packages.go

```diff
@@ -83,12 +83,17 @@ func (p *Packages) Validate(viewName string, absPath string) (*ValidationRespons
 
 	urlPath := "api/v1/packages/analyze?view=" + url.QueryEscape(viewName)
 
+	// #nosec G304
 	fileReader, openErr := os.Open(zipFilePath)
 
 	if openErr != nil {
 		return nil, openErr
 	}
-	defer fileReader.Close()
+	defer func() {
+		if err := fileReader.Close(); err != nil {
+			fmt.Printf("Error closing file: %s\n", err)
```

Comment:
This is not right. We shouldn't let the `api` package write to stdout.

Reply:
Generally I think it's okay to ignore a `Close()` error if it doesn't change the behaviour of the rest. There's no way to handle it in flow, so logging it seems "pointless".

Don't think `Close()` errors that much when you are reading a file. Any read errors will most likely be reflected later in `httpErr`.

Reply:
Lastly, it's "kind of strange" anyway to have an API client function that opens a file, instead of receiving an `io.Reader` and this one even creates a zip file on disk from a directory, which probably doesn't get deleted.

Reply:
Right. I'm mostly just trying to make it stop complaining and ignore the things we already have. That way we hopefully stop introducing more of these. As for logging inside the `api` package, you're right, I've just marked the defer statements such that they are ignored by `gosec`.